src/main/java/org/opentripplanner/framework/geometry/EncodedPolyline.java

```diff
@@ -9,4 +10,8 @@
  * polyline algorithm format</a>
  */
 
-public record EncodedPolyline(String points, int length) implements Serializable {}
+public record EncodedPolyline(String points, int length) implements Serializable {
+  public static EncodedPolyline encode(Geometry geometry) {
```

Comment:
Maybe I don't get it but isn't this against the "aneamic model" principle?

Reply:
Good question! The feature here is mapping from one domain to another. Since we can not change the Geometry (external) so we want can make it simple to find the functionality by inlining it as a static method in EncodedPolyline. But, as you might mean by "aneamic model" - mapping is not the business of EncodedPolyline. So, we take the actual mapping code out into a supporting class `PolylineEncoder`. This way we get the best of both - we provide the users with just one class, and pollute the `EncodedPolyline` with minimal impact.